src/Rules/NoFirstOnSingularModelRule.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Larastan\Larastan\Rules;
+
+use Illuminate\Database\Eloquent\Model;
+use PhpParser\Node;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Identifier;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleError;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPStan\ShouldNotHappenException;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\Type;
+use PHPStan\Type\UnionType;
+
+/** @implements Rule<MethodCall> */
+final class NoFirstOnSingularModelRule implements Rule
+{
+    /**
+     * Specifies the node type this rule is interested in.
+     * We are interested in method calls.
+     *
+     * @return class-string<Node>
+     */
+    public function getNodeType(): string
+    {
+        return MethodCall::class;
+    }
+
+    /**
+     * Processes a MethodCall node to check for redundant ->first() calls.
+     *
+     * @param Node  $node  The current AST node being processed.
+     * @param Scope $scope The current scope.
+     *
+     * @return array<RuleError>
+     *
+     * @throws ShouldNotHappenException
+     */
+    public function processNode(Node $node, Scope $scope): array
+    {
+        if (! $node instanceof MethodCall) {
+            return [];
+        }
```

Comment:
There is no need for this if check---this is guaranteed via the `getNodeType` method